File: jambeth/jambeth-persistence-test/src/main/java/de/osthus/ambeth/model/MaterialGroup.java
package de.osthus.ambeth.model;

import de.osthus.ambeth.annotation.EntityEqualsAspect;
import de.osthus.ambeth.annotation.XmlType;

@XmlType
@EntityEqualsAspect
public class MaterialGroup
{
	protected String id;

	protected short version;

	protected String name;

	protected String buid;

	protected MaterialGroup()
	{
		// Intended blank
	}

	public void setVersion(short version)
	{
		this.version = version;
	}

	public short getVersion()
	{
		return version;
	}

	public void setId(String id)
	{
		this.id = id;
	}

	public String getId()
	{
		return id;
	}

	public void setName(String name)
	{
		this.name = name;
	}

	public String getName()
	{
		return name;
	}

	public void setBuid(String buid)
	{
		this.buid = buid;
	}

	public String getBuid()
	{
		return buid;
	}
}

File: ambeth/Ambeth.Test/ambeth/xml/MaterialGroup.cs
using System;
using De.Osthus.Ambeth.Annotation;

namespace De.Osthus.Ambeth.Test.Model
{
    [XmlType]
    public class MaterialGroup
    {
        public virtual String Id { get; set; }

        public virtual short Version { get; set; }

        public virtual String Name { get; set; }

        public virtual String Buid { get; set; }

        public override bool Equals(Object obj)
        {
            if (obj == this)
            {
                return true;
            }
            if (!(obj is MaterialGroup))
            {
                return false;
            }
            MaterialGroup other = (MaterialGroup)obj;
            bool equals = Id != null ? Id.Equals(other.Id) : false;
            equals &= Version == other.Version;
            return equals;
        }

        public override int GetHashCode()
        {
            return typeof(MaterialGroup).GetHashCode() ^ (Id != null ? Id.GetHashCode() : 1);
        }
    }
}
